### ISSUE TITLE
[valkeylimiter]`Allow()` increments even if not allowed

### ISSUE THREAD

**Issue by th0th** [valkey-io/valkey-go#19]
Hi,

I am using the valkeylimiter for rate limiting. However, even if the limit is reached and `result.Allowed` is `false`, value in valkey is incremented.

```go
func do() error {
	limiter, err := valkeylimiter.NewRateLimiter(valkeylimiter.RateLimiterOption{
		KeyPrefix: "rate_limit",
		Limit:     2,
		Window:    1 * time.Minute,
	})
	if err != nil {
		return err
	}

	// rate_limit:{key} is not set
	result, err := limiter.Allow(context.Background(), "key")
	if err != nil {
		return err
	}
	// rate_limit:{key} is "1"
	result, err = limiter.Allow(context.Background(), "key")
	if err != nil {
		return err
	}
	// rate_limit:{key} is "2"
	result, err = limiter.Allow(context.Background(), "key")
	if err != nil {
		return err
	}
	// rate_limit:{key} is "3"
	fmt.Println(result.Allowed) // false
	// even if the result.Allowed is false, the key is still incremented

	return nil
}
```

**Comment by rueian** [comment]
Hi @th0th,

I think we can add a decrby after: https://github.com/valkey-io/valkey-go/blob/d7c941fc5339b9c4164af00795d3bed7fd207a71/valkeylimiter/limiter.go#L156

Would you like to contribute? we can tag a new alpha release once the fix is merged.

**Comment by th0th** [comment]
Sure, I will send a PR soon!

**Comment by rueian** [comment]
But I also wonder is there any downside if we always increment the counter? Are you worried about  the replication overhead?

**Comment by th0th** [comment]
Not that important but, I thought it would be nice to have the correct number for representing accurately. For example, when displaying as stats somewhere, it will result in something strange like "Login attempts: 224/100".

**Comment by rueian** [comment]
Yes, and please note that we currently rely on the incremented value to determine it is allowed or not in the Go side. If we cap the value, then we may also need to move the check into the lua script.

**Comment by th0th** [comment]
Well, after thinking about it for a while, maybe it is better to leave it as it is now. Because with the current state, it also stores the number of attempts as an additional information. I am closing the issue, thanks for the comments @rueian!
